Lib/test/test_concurrent_futures/test_wait.py

```diff
@@ -114,9 +114,9 @@ def test_all_completed(self):
 
     def test_timeout(self):
         short_timeout = 0.050
-        long_timeout = short_timeout * 10
+        event = threading.Event()
 
-        future = self.executor.submit(time.sleep, long_timeout)
+        future = self.executor.submit(event.wait)
```

Comment:
Just for me, but the reason we do this is because `time.sleep` is not free-threading friendly or is there some other reason?

Reply:
`time.sleep()` tests are generally flaky because they rely on the things happening fast or slow enough and that might not be true if the machine is heavily loaded, or slow, or it's a debug build.

That's not really free threading specific, but removing the GIL changes timing behavior so some flaky tests get more flaky. Other tests probably get *less* flaky, but you're not really going to notice that.